pennylane/ops/qubit/parametric_ops.py

```diff
@@ -276,10 +276,12 @@ def compute_matrix(theta):  # pylint: disable=arguments-differ
         if qml.math.get_interface(theta) == "tensorflow":
             theta = qml.math.cast_like(theta, 1j)
 
-        p = qml.math.exp(-0.5j * theta)
-        z = qml.math.zeros_like(p)
+        arg = 0.5j * theta
+        if qml.math.ndim(arg) == 0:
+            return qml.math.diag(qml.math.exp(arg * np.array([-1, 1])))
```

Comment:
Hi! You can improve performance by distributing arg,
```suggestion
        if qml.math.ndim(arg) == 0:
            return qml.math.diag(qml.math.exp(np.array([-arg, arg])))
```
In my tests this improved timing from:
15.3 µs ± 456 ns per loop
to:
13.8 µs ± 570 ns per loop
This approach can be propagated through the other cases.

Reply:
These benchmarks were made with a single scalar phase (`np.pi`).

Reply:
Nice work on that!